oioswift/proxy/controllers/obj.py

```diff
@@ -213,20 +213,68 @@ def enforce_versioning(self, req):
 
         # We can't use _get_info_from_caches as it would use local worker cache
         # first and an update of versioning mode may not be detected.
-        memcache = getattr(self.app, 'memcache', None) or \
-            req.environ.get('swift.cache')
-        if memcache is None:
+        oio_cache = req.environ.get('oio.cache')
+        memcache = None
+        if oio_cache is None:
+            memcache = getattr(self.app, 'memcache', None) or \
+                req.environ.get('swift.cache')
+
+            if memcache is not None:
+                memcache_key = "/".join(
+                    ("versioning", self.account_name, root_container))
+                version_policy = memcache.get(memcache_key)
```

Comment:
Isn't it possible to put that somewhere in the container sysmeta? This would allow to use the usual cache functions (e.g. `clear_info_cache`).